### PR TITLE
Social Media Icons Widget: add LinkedIn company page support

### LINKED ISSUE #6010: Social Media Icons Widget: Add support for LinkedIn company pages
Enhancement to existing widget: add support to LinkedIn profile link to enable linking to a company instead of an individual LinkedIn user profile.

This is oft-requested by WP.com customers, internal ref 7555-wpcom — at least 15-20 votes so far to add this.

> Currently, the LinkedIn username only works for personal profiles with the URL format https://www.linkedin.com/in/USERNAME
>
> Company pages on LinkedIn have the format https://www.linkedin.com/company/USERNAME
>
> It would be nice to add support for company pages so that business sites can use this widget to link to their company’s page.

This one is tough because this widget is intended to be super-simple, linking to individual user accounts.

We'd need to rewrite it a little bit to allow either:

1) Enter username or full URL (which would help with YouTube and LinkedIn)
2) Use existing Publicize accounts for this user and site to pre-fill the URLs automatically — see ​https://github.com/Automattic/jetpack/issues/2272 and ​https://github.com/Automattic/jetpack/pull/3218

It'd also be nice to add a help link next to each entry field to point to a documentation page on how to get the username or profile URL.

### PR DESCRIPTION
**Pull request by Stoyan0v**
Fixes #6010

#### Changes proposed in this Pull Request:
Add support for company pages in "Social Media Icons" widget.

I think a checkbox below the LinkedIn username field will be a good solution to tell if you are using a company page.

#### Testing instructions:

* Add Social Media Icons widget to sidebar
* Add a company username and check the checkbox below the field.
* Check if the LinkedIn url is changed properly

### CODE REVIEW

File: modules/widgets/social-media-icons.php [hunk 1]
```diff
@@ -240,6 +249,15 @@ class="widefat"
...
 						value="<?php echo esc_attr( $instance[ $service . '_username' ] ); ?>"
 					/>
+					<?php if ( 'linkedin' === $service ): ?>
+						<input
+								class="checkbox"
+								type="checkbox"
+								name="<?php echo $this->get_field_name( $service . '_is_company' ); ?>"
+								<?php checked( $instance[ $service . '_is_company' ], 1 ); ?>
+							/>
+						<label for="<?php echo esc_attr( $this->get_field_id( $service . '_is_company' ) ); ?>">This is a company page.</label>
```
Comment (jeherve) [inline modules/widgets/social-media-icons.php]: Could you make this string translatable with something like `esc_html_e()`?

**Comment by jeherve** [comment]
Also see #6016 for a different approach.

File: modules/widgets/social-media-icons.php (hunk 1 above)
Comment (Stoyan0v) [inline modules/widgets/social-media-icons.php]: Sure, thanks for your comment.

**Comment by georgestephanis** [comment]
I'd personally prefer migrating to the concept behind #6016 as it feels more generally supported, as opposed to manually hacking things in service by service.  We can register multiple schemes for each service (for user vs channel vs company) -- it would involve some migrations to convert over, but it could be handled more generally like some themes have social media menus that operate wholly via css regex on the link url.

**Comment by Stoyan0v** [comment]
@georgestephanis I agree with you that the solution in #6016 is much simpler and will work for more social networks.
I am closing my PR and I am happy that we've found the best fix.

**Pull request closed without merging**